browser/net/brave_torrent_redirect_network_delegate_helper.cc

```diff
@@ -50,28 +54,23 @@ int OnHeadersReceived_TorrentRedirectWork(
     GURL* allowed_unsafe_redirect_url,
     const brave::ResponseCallback& next_callback,
     std::shared_ptr<brave::BraveRequestInfo> ctx) {
-
-  if (!original_response_headers ||
-      !IsMainFrameResource(ctx) ||
+  if (!original_response_headers || !IsMainFrameResource(ctx) ||
       ctx->is_webtorrent_disabled ||
       // download .torrent, do not redirect
       (IsWebtorrentInitiated(ctx) && !IsViewerURL(ctx->request_url)) ||
       !IsTorrentFile(ctx->request_url, original_response_headers)) {
     return net::OK;
   }
 
-  *override_response_headers =
-    new net::HttpResponseHeaders(original_response_headers->raw_headers());
-  (*override_response_headers)
-      ->ReplaceStatusLine("HTTP/1.1 307 Temporary Redirect");
-  (*override_response_headers)->RemoveHeader("Location");
-  GURL url(
-      base::StrCat({extensions::kExtensionScheme, "://",
-      brave_webtorrent_extension_id,
-      "/extension/brave_webtorrent2.html?",
-      ctx->request_url.spec()}));
-  (*override_response_headers)->AddHeader("Location", url.spec());
-  *allowed_unsafe_redirect_url = url;
+  auto* contents =
+      content::WebContents::FromFrameTreeNodeId(ctx->frame_tree_node_id);
+  GURL url(base::StrCat(
+      {extensions::kExtensionScheme, "://", brave_webtorrent_extension_id,
+       "/extension/brave_webtorrent2.html?", ctx->request_url.spec()}));
+  NavigateParams params(chrome::FindBrowserWithWebContents(contents), url,
+                        ui::PAGE_TRANSITION_FROM_ADDRESS_BAR);
```

Comment:
@diracdeltas do you know if there are any other security implications to `ui::PAGE_TRANSITION_FROM_ADDRESS_BAR`? This seems to work quite well.

Reply:
i'm not sure if there is a better API for loading a URL in an existing tab. my first thought was whatever https://developer.chrome.com/docs/extensions/reference/tabs/#type-Tab uses under the hood. 

`ui::PAGE_TRANSITION_FROM_ADDRESS_BAR` may have some security implications since it implies a user gesture, which is not really the case here.

Reply:
looks like that is here: https://source.chromium.org/chromium/chromium/src/+/main:chrome/browser/extensions/api/tabs/tabs_api.cc;l=1675. and the other params are probably important to set here too.